bigquery_etl/shredder/config.py

```diff
@@ -486,35 +466,63 @@ def find_glean_targets(
         for table in tables
         if table.labels.get("schema_id") == GLEAN_SCHEMA_ID
     ]
+
+    channel_to_app_name = get_glean_channel_to_app_name_mapping()
+
+    # create mapping of dataset -> (tables containing associated deletion requests)
     # construct values as tuples because that is what they must be in the return type
     sources: dict[str, tuple[DeleteSource, ...]] = defaultdict(tuple)
+    app_names = set()
     source_doctype = "deletion_request"
     for table in glean_stable_tables:
         if table.table_id.startswith(source_doctype):
             source = DeleteSource(qualified_table_id(table), GLEAN_CLIENT_ID, project)
-            derived_dataset = re.sub("_stable$", "_derived", table.dataset_id)
+
+            channel_name = re.sub("_stable$", "", table.dataset_id)
+            derived_dataset = channel_name + "_derived"
+            app_name = channel_to_app_name.get(channel_name)
+
             # append to tuple to use every version of deletion request tables
             sources[table.dataset_id] += (source,)
             sources[derived_dataset] += (source,)
-    glean_derived_tables = list(
-        pool.map(
+
+            if app_name is not None and app_name != channel_name:
+                app_names.add(app_name)
+                sources[app_name + "_derived"] += (source,)
+
+    # use deletion request view containing all channels if found
```

Comment:
Why do these views exist only for some apps? How were they created?

Reply:
This for apps that are ignored by glean_usage sql generation listed in: https://github.com/mozilla/bigquery-etl/blob/0ec741c910ebffed6e3f96807bf2d55c7b70deca/bqetl_project.yaml#L428-L434